segmentstore/server/src/main/java/io/pravega/segmentstore/server/security/TLSConfigChangeEventConsumer.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.security;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import io.netty.handler.ssl.SslContext;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A {@link Consumer} that acts upon change in Segment Store SSL/TLS configuration change.
+ */
+@RequiredArgsConstructor
+@Slf4j
+public class TLSConfigChangeEventConsumer implements Consumer<WatchEvent<?>> {
+
+    /**
+     * A counter representing the number of times this object has been asked to
+     * consume an event.
+     */
+    private final AtomicInteger numOfConfigChangesSinceStart = new AtomicInteger(0);
+
+    private @NonNull final AtomicReference<SslContext> sslContext;
+    private @NonNull final String pathToCertificateFile;
+    private @NonNull final String pathToKeyFile;
+
+
+    @Override
+    public void accept(WatchEvent<?> watchEvent) {
+        log.debug("Invoked for [{}]", watchEvent.context());
+        handleTlsConfigChange();
+    }
+
+    private void handleTlsConfigChange() {
+        log.info("Current reload count = {}", numOfConfigChangesSinceStart.incrementAndGet());
+        sslContext.set(TLSHelper.newServerSslContext(pathToCertificateFile, pathToKeyFile));
```

Comment:
This looks good, but please add some robustness here.   I believe that a watch event may come while the updated file is still being written, leading to an IOException (i.e. a race).   Please catch IOException here and schedule a retry.

Reply:
According to the [Java Doc](https://docs.oracle.com/javase/8/docs/api/java/nio/file/StandardWatchEventKinds.html#ENTRY_MODIFY): 
* ENTRY_MODIFY represents "Directory entry modified"
* ENTRY_CREATE represents "Directory entry created."

The `WatchDir` [example](https://docs.oracle.com/javase/tutorial/essential/io/examples/WatchDir.java) from Oracle also doesn't do anything special in this respect.

My understanding is that these events are raised only after the corresponding operation is done. For the `ENTRY_MODIFY` event, multiple events might be raised, usually once for content and the second time for metadata. I've added a sleep-based deduplication logic in  `FileModificationWatcher.java` to compensate for it. 

That being said, your comment prompted me to add some additional checks in `TLSHelper.newServerSslContext(...)` to ensure that the files exist and are readable.